### PR TITLE
chore: remove unused code related to `random_number`

### PR DESCRIPTION
**Pull request by Pranavchiku**
Addresses https://github.com/lfortran/lfortran/pull/3754#issuecomment-2030038128

### CODE REVIEW

**Comment by certik** [comment]
I have a question --- what about user defined elemental subroutines?

We need to support this feature it seems. Let's instead add tests for user elemental subroutines and ensure things work.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
Remove it from LPython as well.

**Comment by certik** [comment]
@czgdp1807 but why removing in the first place? I think we need this functionality for user defined elemental subroutines.

**Comment by Pranavchiku** [comment]
I remember we had a discussion somewhere about this, will check and circle back here.

**Comment by Pranavchiku** [comment]
Elemental subroutine are not broadcasted yet, we do not support it. Issue: https://github.com/lfortran/lfortran/issues/3730, this will not be difficult to fix.

File: src/libasr/pass/array_op.cpp [hunk 1]
```diff
@@ -1859,163 +1859,6 @@ class ArrayOpVisitor : public ASR::CallReplacerOnExpressionsVisitor<ArrayOpVisit
...
-
-        }
-
-        void visit_SubroutineCall(const ASR::SubroutineCall_t& x) {
-            ASR::symbol_t* sym = x.m_original_name;
-            if (sym && ASR::is_a<ASR::ExternalSymbol_t>(*sym)) {
-                ASR::ExternalSymbol_t* ext_sym = ASR::down_cast<ASR::ExternalSymbol_t>(sym);
-                std::string name = ext_sym->m_name;
-                std::string module_name = ext_sym->m_module_name;
-                if (module_name == "lfortran_intrinsic_math" && name == "random_number") {
```
Comment (Pranavchiku) [inline src/libasr/pass/array_op.cpp:1960]: Instead of this, one has to check if call is to elemental subroutine, if yes, same code for broadcast.

**Comment by gxyd** [comment]
I just saw the discussion here, the original PR I opened (i.e. #3754) was only to check if tests succeed without the existing code, I didn't actually intend to remove it, as mentioned above by @Pranavchiku , I might be able to re-use it for implementing #3730 .

**Comment by Pranavchiku** [comment]
closing with reference to #3730.

**Pull request closed without merging**